adaptive/src/main/java/com/google/accompanist/adaptive/FoldAwareColumn.kt

```diff
@@ -0,0 +1,1247 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.accompanist.adaptive
+
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Immutable
+import androidx.compose.runtime.Stable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Rect
+import androidx.compose.ui.graphics.toComposeRect
+import androidx.compose.ui.layout.AlignmentLine
+import androidx.compose.ui.layout.IntrinsicMeasurable
+import androidx.compose.ui.layout.IntrinsicMeasureScope
+import androidx.compose.ui.layout.Layout
+import androidx.compose.ui.layout.Measurable
+import androidx.compose.ui.layout.MeasurePolicy
+import androidx.compose.ui.layout.MeasureResult
+import androidx.compose.ui.layout.MeasureScope
+import androidx.compose.ui.layout.Measured
+import androidx.compose.ui.layout.ParentDataModifier
+import androidx.compose.ui.layout.Placeable
+import androidx.compose.ui.layout.boundsInWindow
+import androidx.compose.ui.platform.InspectorInfo
+import androidx.compose.ui.platform.InspectorValueInfo
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.unit.Constraints
+import androidx.compose.ui.unit.Density
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.LayoutDirection
+import androidx.compose.ui.util.fastForEach
+import androidx.window.layout.DisplayFeature
+import androidx.window.layout.FoldingFeature
+import com.google.accompanist.adaptive.LayoutOrientation.Horizontal
+import com.google.accompanist.adaptive.LayoutOrientation.Vertical
+import kotlin.math.max
+import kotlin.math.min
+import kotlin.math.roundToInt
+import kotlin.math.sign
+
+/**
+ * A simplified version of [Column] that places children in a fold-aware manner.
+ *
+ * @param displayFeatures the list of known display features to automatically avoid
```

Comment:
I would change the param description to: a list of display features the device currently has, as that appears to be what is passed in. 

We will avoid the ones that are relevant to our design not all of them. In the logic, we only avoid Horizontal Folds. Any other display features are not considered, so the description is not fully accurate. 

I would just say it is a list of display features that the device currently has. 

However, I also think this might be an unnecessary parameter. It should be done within the code logic without requiring the developer to get this information. However, I think for Accompanist we can have this here.

Reply:
Great catch, just updated the parameter description!

And that's a good point about whether the parameter is necessary - I'll leave it for now like you said since it matches the `TwoPane` constructor for Accompanist, but we can think about how to remove it going forward.